### PR TITLE
Use Lister to reduce the number of calls to api server

### PR DESCRIPTION
**Pull request by elgnay**
The related issue: https://github.com/open-cluster-management/backlog/issues/8687

Signed-off-by: Yang Le <yangle@redhat.com>

### CODE REVIEW

File: pkg/spoke/managedcluster/creating_controller.go [hunk 1]
```diff
@@ -43,9 +44,13 @@ func NewManagedClusterCreatingController(
 		spokeCABundle:           spokeCABundle,
 		hubClusterClient:        hubClusterClient,
 	}
+
+	// create a random interval between CreatingControllerSyncInterval and CreatingControllerSyncInterval * 2.
+	interval := CreatingControllerSyncInterval + time.Duration(rand.Float64()*float64(CreatingControllerSyncInterval))
```
Comment (qiujian16) [inline pkg/spoke/managedcluster/creating_controller.go]: could use wait.Jitter I think

Comment (elgnay) [reply pkg/spoke/managedcluster/creating_controller.go]: Updated.

**Review by qiujian16 — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**